### PR TITLE
Addressing Klocwork issues:

### PR DESCRIPTION
**Pull request by asgibbons**
1. Dereference 'ws' which may be NULL.
2. Free tracker info when open returns.

### CODE REVIEW

**Comment by michael-adler** [comment]
Wouldn't it be better to check that wm isn't NULL after line 203 and return an error status in that case? You'd only have to change one line and the error return from find_or_map_wm() would return to being meaningful.

**Comment by asgibbons** [comment]
@michael-adler I'm confused by your comment.  `result` will be set as the return value from the function, which also returns `wm`.  If `result` is non-zero or `wm` is `NULL`, then the function returns `result` - the return value from `find_or_map_wm()`.  Isn't that what is desired here?

**Comment by michael-adler** [comment]
@asgibbons I now understand the confusion. No, find_or_map_wm() should never return NULL when result is non-zero. When wsid_find_by_index() returns NULL, map_mmio_region() is called. This is supposed to then guarantee that wsid_find_by_index() returns a non-NULL result. As a result, you should be able to simply confirm that wsid_find_by_index() returns non-NULL and, if it does, raise some error.

**Comment by asgibbons** [comment]
@michael-adler I finally see your point.  Thanks for the explanation. 👍

**Comment by asgibbons** [comment]
@OPAE/api Can I get an approval please?

**Comment by asgibbons** [comment]
@anandaravuri Can I get an approval please?

File: libopae/src/mmio.c [hunk 1]
```diff
@@ -201,6 +201,10 @@ static fpga_result find_or_map_wm(fpga_handle handle, uint32_t mmio_num,
 			return result;
 		}
 		wm = wsid_find_by_index(_handle->mmio_root, mmio_num);
+		if (!wm) {
+			FPGA_ERR("unable to map wsid for mmio region %d", mmio_num);
+			return FPGA_NO_MEMORY;
```
Comment (r-rojo) [inline libopae/src/mmio.c]: Good fix. Would it make sense to return FPGA_EXCEPTION or FPGA_NOT_FOUND?

Comment (asgibbons) [reply libopae/src/mmio.c]: I think NO_MEMORY is better since this is effectively a memory allocation.

**Pull request merged**